sdk/file_sdk/OPCFileFormat/zip_file.h

```diff
@@ -0,0 +1,87 @@
+#ifndef ZIPAPI_H
+#define ZIPAPI_H
+
+#include "common/exceptions.h"
+#include "common/gsf_input_istream.h"
+#include "Common/istream.h"
+#include <map>
+#include <string>
+#include <memory>
+#include <gsf/gsf.h>
+
+namespace mip {
+namespace file {
+
+struct GsfOutput_deleter {
+  void operator () (GsfOutput* obj) const {
+    if (!gsf_output_is_closed(obj)) {
+      gsf_output_close(obj);
+    }
+    g_object_unref(G_OBJECT(obj));
+  }
+};
+
+struct GsfOutfile_deleter {
+  void operator () (GsfOutfile* obj) const {
+    if (!gsf_output_is_closed(GSF_OUTPUT(obj))) {
+      gsf_output_close(GSF_OUTPUT(obj));
+    }
+    g_object_unref(G_OBJECT(obj));
+  }
+};
+
+struct GsfInput_deleter {
+  void operator () (GsfInput* obj) const {
+    g_object_unref(G_OBJECT(obj));
+  }
+};
+
+struct GsfInfile_deleter {
+  void operator () (GsfInfile* obj) const {
+    g_object_unref(G_OBJECT(obj));
+  }
+};
+
+struct GsfStrSplit_deleter {
+  void operator () (gchar** obj) const {
+    g_strfreev(obj);
+  }
+};
+
+struct GError_deleter {
+  void operator () (GError* obj) const {
+    if(obj != nullptr)
+      g_error_free(obj);
+  }
+};
+
+class ZipFile {
+
+public:
+  ZipFile(std::shared_ptr<IStream> inputStream);
+  std::string GetEntry(const std::string& entryPath);
+  void SetEntry(const std::string& entryPath, const std::string& content);
+  void Commit(std::shared_ptr<IStream> outputStream);
+
+protected:
```

Comment:
>protected [](start = 0, length = 9)

private

Reply:
and remove space after

---
In reply to: [139358243](https://github.com/AzureAD/rms-sdk-for-cpp/pull/158#discussion_r139358243) [](ancestors = 139358243)